src/applications/auth/containers/AuthApp.jsx

```diff
@@ -94,6 +98,42 @@ export default function AuthApp({ location }) {
     setHasError(true);
   };
 
+  const handleProvisioning = async userProfile => {
+    const { signIn } = userProfile;
+    if (
+      signIn?.ssoe &&
+      returnUrl.includes(EXTERNAL_REDIRECTS[EXTERNAL_APPS.MY_VA_HEALTH])
```

Comment:
I think this thing should come out so it'll be like

```
if(returnUrl.includes(EXTERNAL_REDIRECTS[EXTERNAL_APPS.MY_VA_HEALTH])) {
  await handleProvisioning(userProfile);
}
```

I have to pull it down again to look at ESLint again (it complains about arrow functions and returns if I remember correctly) but I'd rather just return `undefined` since its a side-effect and we don't do anything with the true/false.

Great job overall 🚀

Reply:
Thank you! Happy to make those adjustments.
Right now it returns true/false in order to execute an early return in the `handleAuthSuccess` function if the user is not provisioned:
```jsx
const { userProfile } = authMetrics;
const provisioned = await handleProvisioning(userProfile);
if (!provisioned) {
  return;
}
```
Do you want me to get rid of that, return undefined, and just have it redirect regardless if the user is provisioned or not?

Reply:
@ianmagenta 

Yeah I think if we change the `handleProvisioning` to be a regular function (non-arrow function) we won't get the lint error about arrow functions and consistent returns.

```
async function handleProvisioning() {
    try {
      const termsResponse = await apiRequest(
        `/terms_of_use_agreements/update_provisioning`,
        {
          method: 'PUT',
          headers: { 'Content-Type': 'application/json' },
          credentials: 'include',
        },
      );
      if (!termsResponse?.provisioned) {
        handleAuthError(null, '111');
      }
    } catch (err) {
      const message = err?.error;
      if (message === 'Agreement not accepted') {
        window.location = `${
          environment.BASE_URL
        }/terms-of-use/myvahealth/?ssoeTarget=${returnUrl}`;
      } else if (message === 'Account not Provisioned') {
        handleAuthError(null, '111');
      } else {
        handleAuthError(err, '110');
      }
    }
  }
  ```
  
Then in the `handleAuthSuccess` we can do this:
  
```
if(returnUrl.includes(myvahealth_variable) && userProfile?.signIn?.ssoe) {
  await handleProvisioning();
}
```

Only thing I'm not sure about is in the `handleProvisioning` function when the `Agreement not accepted` whether we should send a user to the Terms of Use page (I'd ask Joe or Trevor, my guess is yes but you never know 🤷 ). Thank you for your patience with this.